packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../auto_dispose_mixin.dart';
+import '../ui/search.dart';
+import '../utils.dart';
+import 'debugger_controller.dart';
+import 'debugger_model.dart';
+
+const int numOfMatchesToShow = 6;
+
+class FileSearchField extends StatefulWidget {
+  const FileSearchField({
+    @required this.controller,
+    @required this.handleClose,
+  });
+
+  final DebuggerController controller;
+  final Function handleClose;
+
+  @override
+  _FileSearchFieldState createState() => _FileSearchFieldState();
+}
+
+class _FileSearchFieldState extends State<FileSearchField>
+    with SearchFieldMixin, AutoDisposeMixin {
+  AutoCompleteController _autoCompleteController;
+  final Map<String, ScriptRef> _scriptsCache = {};
+
+  final fileSearchFieldKey = GlobalKey(debugLabel: 'fileSearchFieldKey');
+
+  @override
+  void initState() {
+    super.initState();
+
+    _autoCompleteController = AutoCompleteController();
+    _autoCompleteController.currentDefaultIndex = 0;
+
+    addAutoDisposeListener(
+        _autoCompleteController.selectTheSearchNotifier, _handleSearch);
+    addAutoDisposeListener(
+        _autoCompleteController.searchNotifier, _handleSearch);
+
+    _autoCompleteController.selectTheSearch = true;
+  }
+
+  void _handleSearch() {
+    final query = _autoCompleteController.search;
+    final matches = findMatches(query, widget.controller.sortedScripts.value);
+    matches.forEach(_addScriptRefToCache);
+    _autoCompleteController.searchAutoComplete.value =
+        matches.map((scriptRef) => scriptRef.uri).toList();
+    _handleAutoCompleteOverlay();
+  }
+
+  void _handleAutoCompleteOverlay() {
+    _autoCompleteController.handleAutoCompleteOverlay(
+      context: context,
+      searchFieldKey: fileSearchFieldKey,
+      onTap: _onSelection,
+    );
+  }
+
+  void _addScriptRefToCache(ScriptRef scriptRef) {
+    if (!_scriptsCache.containsKey(scriptRef.uri)) {
+      _scriptsCache[scriptRef.uri] = scriptRef;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Container(
+      decoration: BoxDecoration(
+        border: Border(
+          top: BorderSide(color: theme.focusColor),
+        ),
+      ),
+      padding: const EdgeInsets.all(8.0),
+      child: Row(
+        children: [
+          const SizedBox(width: 8.0),
+          Expanded(
+            child: buildAutoCompleteSearchField(
+              controller: _autoCompleteController,
+              searchFieldKey: fileSearchFieldKey,
+              searchFieldEnabled: true,
+              shouldRequestFocus: true,
+              closeOverlayOnEscape: false,
```

Comment:
This skips closing the autocomplete overlay. So ESC will close the dialog (otherwise a user first has to click ESC to close the autocomplete results, then click ESC again to close the dialog)